src/SourcemapToolkit.SourcemapParser/SourceMap.cs

```diff
@@ -112,12 +113,48 @@ public SourceMap ApplySourceMap(SourceMap submap, string sourceFile = null)
 			return newSourceMap;
 		}
 
-		/// <summary>
-		/// Finds the mapping entry for the generated source position. If no exact match is found, it will attempt
-		/// to return a nearby mapping that should map to the same piece of code.
-		/// </summary>
-		/// <param name="generatedSourcePosition">The location in generated code for which we want to discover a mapping entry</param>
-		/// <returns>A mapping entry that is a close match for the desired generated code location</returns>
+        /// <summary>
+        /// Removes column information from a source map
+        /// This can significantly reduce the size of source maps, if line numbers are correct
+        /// <returns>A new source map</returns>
+        /// </summary>
+        public SourceMap Flatten()
```

Comment:
good point. I didn't want to mutate the SourceMap state, so I wanted to make a new one